crates/network/src/lib.rs

```diff
@@ -33,24 +34,22 @@ pub use any::AnyNetwork;
 
 pub use alloy_eips::eip2718;
 
-/// A list of transactions, either hydrated or hashes.
-#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
-#[serde(untagged)]
-pub enum TransactionList<T> {
-    /// Hashes only.
-    Hashes(Vec<B256>),
-    /// Hydrated tx objects.
-    Hydrated(Vec<T>),
-    /// Special case for uncle response
-    Uncled,
+/// A block response.
+pub trait BlockResponse<N: Network> {
```

Comment:
here we can also integrate a few more helper functions that currently exist on block, for iterating over txs for example.